cli/src/main/java/com/devonfw/tools/ide/commandlet/UpgradeCommandlet.java

```diff
@@ -0,0 +1,179 @@
+   *
+   * First compares base versions (e.g. 2024.12.002 with 2025.01.001), then timestamps if base versions are equal.
+   * Returns false if version formats are unexpected to avoid unintended upgrades.
+   *
+   * @param currentVersion The current snapshot version
+   * @param latestVersion The latest snapshot version to compare against
+   * @return true if latestVersion is newer than currentVersion, false otherwise or if formats are invalid
+   */
+  protected boolean isSnapshotNewer(String currentVersion, String latestVersion) {
+
+    try {
+      // Validate input formats
+      if (currentVersion == null || latestVersion == null || !currentVersion.contains("-") || !latestVersion.contains(
+          "-")) {
+        return false;
+      }
+
+      // First compare base versions (2024.12.002 with 2025.01.001)
+      String currentBase = currentVersion.substring(0, currentVersion.indexOf('-'));
+      String latestBase = latestVersion.substring(0, latestVersion.indexOf('-'));
+
+      VersionIdentifier currentBaseVersion = VersionIdentifier.of(currentBase);
+      VersionIdentifier latestBaseVersion = VersionIdentifier.of(latestBase);
+
+      // If base versions are different, use regular version comparison
+      if (!currentBaseVersion.compareVersion(latestBaseVersion).isEqual()) {
+        return currentBaseVersion.compareVersion(latestBaseVersion).isLess();
+      }
+
+      // Validate timestamp formats
+      String[] currentParts = currentVersion.split("-");
+      String[] latestParts = latestVersion.split("-");
+      if (currentParts.length < 3 || latestParts.length < 3) {
+        return false;
+      }
+
+      // Extract timestamps
+      String currentTimestamp = currentParts[2].split("-")[0].replace("_", ""); // "010102"
+      String[] latestTimestampParts = latestParts[2].split("\\.");
+      if (latestTimestampParts.length < 1) {
+        return false;
+      }
+
+      // Get year from base version (2024.12.002 -> 2024)
+      String year = currentBase.substring(0, 4);
+
+      // Parse current date/time using extracted year (currentTimestamp format: MMDDXX)
+      LocalDateTime currentTime = LocalDateTime.parse(year + currentTimestamp + "00", // YYYYMMDDHHmm
+          DateTimeFormatter.ofPattern("yyyyMMddHHmm"));
+
+      // Parse latest date/time (format: YYYYMMDD.HHMMSS)
+      LocalDateTime latestTime = LocalDateTime.parse(latestTimestampParts[0] + "000000",
+          DateTimeFormatter.ofPattern("yyyyMMddHHmmss"));
+
+      return latestTime.isAfter(currentTime);
+    } catch (Exception e) {
+      throw new IllegalStateException("Failed to compare current and latest Snapshot.", e);
+    }
+  }
+
+  @Override
+  public void run() {
+
+    ToolRepository mavenRepo = this.context.getMavenSoftwareRepository();
+    VersionIdentifier currentVersion = VersionIdentifier.of(IdeVersion.get());
+    VersionIdentifier latestVersion = mavenRepo.resolveVersion(IDEASY_GROUP_ID, IDEASY_ARTIFACT_ID, currentVersion);
```

Comment:
why does `resolveVersion` give us the latest version?
According to my understanding it will resolve version patterns or SNAPSHOTs to the latest matching version number.
If we put `2024.01.001` in as `currentVersion`, then according to the JavaDoc you should also get `2024.01.001` what is not the `latestVersion`.
>    @return the resolved {@link VersionIdentifier}. If the given {@link VersionIdentifier} is NOT a {@link VersionIdentifier#isPattern() pattern} this method
     will always just return the given {@link VersionIdentifier}.